feature/onboarding/permissions/src/main/kotlin/app/k9mail/feature/onboarding/permissions/ui/PermissionsViewModel.kt

```diff
@@ -0,0 +1,112 @@
+package app.k9mail.feature.onboarding.permissions.ui
+
+import androidx.lifecycle.viewModelScope
+import app.k9mail.core.ui.compose.common.mvi.BaseViewModel
+import app.k9mail.feature.onboarding.permissions.domain.PermissionsDomainContract.UseCase
+import app.k9mail.feature.onboarding.permissions.domain.entity.PermissionState
+import app.k9mail.feature.onboarding.permissions.ui.PermissionsContract.Effect
+import app.k9mail.feature.onboarding.permissions.ui.PermissionsContract.Event
+import app.k9mail.feature.onboarding.permissions.ui.PermissionsContract.State
+import app.k9mail.feature.onboarding.permissions.ui.PermissionsContract.UiPermissionState
+import app.k9mail.feature.onboarding.permissions.ui.PermissionsContract.ViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class PermissionsViewModel(
+    private val checkPermission: UseCase.CheckPermission,
+    private val backgroundDispatcher: CoroutineDispatcher = Dispatchers.IO,
+) : BaseViewModel<State, Event, Effect>(initialState = State(isLoading = true)), ViewModel {
+
+    override fun event(event: Event) {
+        when (event) {
+            Event.LoadPermissionState -> loadPermissionState()
+            Event.AllowContactsPermissionClicked -> handleAllowContactsPermissionClicked()
+            Event.AllowNotificationsPermissionClicked -> handleAllowNotificationsPermissionClicked()
+            is Event.ContactsPermissionResult -> handleContactsPermissionResult(event.success)
+            is Event.NotificationsPermissionResult -> handleNotificationsPermissionResult(event.success)
+            Event.NextClicked -> handleNextClicked()
+        }
+    }
+
+    private fun loadPermissionState() {
+        if (!state.value.isLoading) {
+            // TODO: Use handleOneTimeEvent() once available to prevent multiple initializations
```

Comment:
This is available now

Reply:
Nice catch. Fixed in 7a249fb42e83e4913fdbe132fb738e4cf29e4a73